csrc/python_frontend/fusion_record.h

```diff
@@ -380,6 +381,79 @@ struct ReshapeOpRecord : RecordFunctor {
   std::vector<int64_t> new_shape_;
 };
 
+struct PadOpRecord : RecordFunctor {
+  PadOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      std::vector<int64_t>& pad_widths)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.pad",
+            RecordType::PadOp),
+        pad_widths_(std::move(pad_widths)) {}
+  virtual ~PadOpRecord() = default;
+  virtual RecordFunctor* clone() final {
+    return new PadOpRecord(*this);
+  }
+
+  virtual size_t hash() const final {
+    auto result = RecordFunctor::hash();
+    size_t widths_hash = 0;
+    // for (auto w : pad_widths_) {
+    // TODO: How to hash vector<Scalar*> pad_widths here?
+    // widths_hash ^= w.hash();
+    //}
+    return result | (widths_hash & 0xffff);
+  }
+
+  /*
+  virtual bool operator==(const RecordFunctor& other) const final {
+    auto result = false;
+    if (auto child_ptr = dynamic_cast<const PadOpRecord*>(&other)) {
+      result = RecordFunctor::operator==(other);
+    }
+    return result;
+  }
+  */
+
+  void operator()(FusionState& fd) final {
+    auto arg = fd.getFusionState(args_.at(0).index)->template as<TensorView>();
+    auto fill_value = fd.getFusionState(args_.at(1).index);
+    std::vector<Val*> val_widths;
+    val_widths.reserve(pad_widths_.size());
+    for (auto p : pad_widths_) {
+      auto pval = IrBuilder::create<Int>(p);
+      val_widths.push_back(pval);
+    }
+    auto output = pad(arg, val_widths, fill_value);
+    fd.setFusionState(outputs_.at(0).index, output);
+  }
+
+  void print(std::ostream& os, bool close_function = true) const final {
+    RecordFunctor::print(os, false);
+    os << ", pad_widths=[";
+    bool first_arg = true;
+    for (auto w : pad_widths_) {
+      if (first_arg) {
+        first_arg = false;
+      } else {
+        os << ", ";
+      }
+      os << w;
+    }
+    os << "]";
+    if (close_function) {
+      os << ")";
+    }
+  }
+
+ private:
+  //! Pairs of non-negative integers indicating the amount to pad the front and
+  //! back of each dimension.
+  std::vector<int64_t> pad_widths_;
```

Comment:
pad_widths_ are now static on FusionRecord, we need to handle them in cache, otherwise you could ended up hitting the wrong cached fusion.